lib/toolkit/loader.js

```diff
@@ -431,9 +444,14 @@ const nodeResolve = iced(function nodeResolve(id, requirer, { rootURI }) {
   if ((resolvedPath = loadAsDirectory(fullId)))
     return stripBase(rootURI, resolvedPath);
 
+  // If the requirer is an absolute URI then the node module resolution below
+  // won't work correctly as we prefix everything with rootURI
+  if (isAbsoluteURI(requirer))
+    return void 0;
```

Comment:
It's a weird case and it's possible it only appears in tests, but at the very least test-addon-bootstrap triggers this. The way it loads the code here https://github.com/mozilla/addon-sdk/blob/master/test/test-addon-bootstrap.js#L37 means we end up hitting this case with the require calls from `sdk/addon/bootstrap` but only when it requires non-relative paths.